### PR TITLE
`1.87` cherry-picks

### PR DESCRIPTION
**Pull request by Duciwuci**
Here are the cherry-picks on our side for the `v1.87` YAKE release :) 

Based on latest https://github.com/YAKEcloud/yake/releases/tag/v1.87.4-0

### CODE REVIEW

File: gardener/etcd-events.yaml [hunk 1]
```diff
@@ -74,7 +74,16 @@ spec:
...
                 value: gardener-kubernetes
+              - op: replace
+                path: /spec/template/spec/containers/0/volumeMounts/0/name
+                value: etcd-events
+              - op: replace
+                path: /spec/template/spec/containers/1/volumeMounts/1/name
+                value: etcd-events
+              - op: replace
+                path: /spec/volumeClaimTemplates/0/metadata/name
+                value: etcd-events
```
Comment (nschad) [inline gardener/etcd-events.yaml:89]: Honestly I would prefer if we could just rename the volume. I feel like these /0/ positional stuff is just bound to break

File: helmcharts/etcd/templates/secret-etcd-backup.yaml [hunk 2]
```diff
@@ -19,6 +19,8 @@ metadata:
   name: {{ .Values.name }}-backup
   namespace: {{ .Release.Namespace }}
 type: Opaque
+stringData:
+{{ toYaml .Values.backup.secretStringData | indent 2 }}
```
Comment (nschad) [inline helmcharts/etcd/templates/secret-etcd-backup.yaml]:
```suggestion
{{- toYaml .Values.backup.secretStringData | nindent 2 }}
```
(strips whitespaces first, then new-line idents by 2)

let's be more error-resistant

File: gardener/etcd-events.yaml (hunk 1 above)
Comment (nschad) [reply gardener/etcd-events.yaml:89]: Besides we will never be able to get rid of it.

**Comment by Duciwuci** [comment]
/hold we need another fix in yake

`etcd-events` is using the community charts and we need to point it to the local chart

File: gardener/etcd-events.yaml (hunk 1 above)
Comment (maboehm) [reply gardener/etcd-events.yaml:89]: we are planning on contributing a new option to the helmchart like `volumeNameOverride`. Then this commit can be dropped.

And before we just rename it, id rather work on HA vgarden first, which will porbably result in new volumes anyways

Comment (nschad) [reply gardener/etcd-events.yaml:89]: okay 👍

**Review by nschad — APPROVED** [review]
/lgtm

**Pull request merged**